### PR TITLE
MOTECH-2939: Fixed blank/null values in Observation of the same type

### PR DESCRIPTION
**Pull request by MichalKus**
Now if value is blank/null Observation is not created.

### CODE REVIEW

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

File: openmrs/src/main/java/org/motechproject/openmrs/tasks/impl/OpenMRSActionProxyServiceImpl.java [hunk 1]
```diff
@@ -266,18 +266,20 @@ private Location getLocationByName(String configName, String locationName) {
             if (valueIsNotEmpty(observations, observationConceptName)) {
                 String[] observationValues = observations.get(observationConceptName).replace(", ", ",").split(",");
                 for (String value : observationValues) {
-                    Observation observation = new Observation();
+                    if (value != null && !value.isEmpty()) {
```
Comment (kmadej) [inline openmrs/src/main/java/org/motechproject/openmrs/tasks/impl/OpenMRSActionProxyServiceImpl.java]: StringUtils.isNotEmpty

File: openmrs/src/test/java/org/motechproject/openmrs/tasks/impl/OpenMRSActionProxyServiceTest.java [hunk 2]
```diff
@@ -190,7 +190,7 @@ public void shouldCreateEncounterWithGivenParametersWithObsWithManyValues() {
 
         DateTime encounterDatetime = new DateTime("2000-08-16T07:22:05Z");
         Map<String, String> observations = new HashMap<>();
-        observations.put("testConceptName","testObservationValueName0, testObservationValueName1");
+        observations.put("testConceptName","testObservationValueName0, ,testObservationValueName1");
```
Comment (kmadej) [inline openmrs/src/test/java/org/motechproject/openmrs/tasks/impl/OpenMRSActionProxyServiceTest.java:194]: it easy to lose this empty value, can you add a brief java comment that empty value should be not included

**Review by kmadej — CHANGES_REQUESTED** [review]

**Review by kmadej — APPROVED** [review]

**Pull request merged**